src/services/MessagePreviewService.ts

```diff
@@ -23,31 +25,59 @@ class MessagePreviewService {
 				const channel = callingMessage.guild.channels.cache.get(msgArray[1]) as TextChannel;
 				const messageToPreview = await channel.messages.fetch(msgArray[2]);
 
-				if (!this.wasSentByABot(messageToPreview)) {
+				if (!messageToPreview.author?.bot) {
 					const embed = new MessageEmbed();
+					const parsedContent = this.escapeHyperlinks(messageToPreview.content);
 
-					embed.setAuthor(messageToPreview.member?.nickname || messageToPreview.author.username, messageToPreview.author.avatarURL() || undefined, link);
-					embed.addField(`Called by ${callingMessage.member?.nickname || callingMessage.author.username}`, `[Click for context](${link})`);
-					embed.setDescription(`${messageToPreview.content}\n`);
-					embed.setColor(messageToPreview.member?.displayColor || "#FFFFFE");
+					embed.setAuthor(this.getAuthorName(messageToPreview), messageToPreview.author.avatarURL() || undefined, link);
+					embed.setDescription(`**#${this.getChannelName(messageToPreview)}**\n\n${parsedContent}\n`);
+					embed.addField(FIELD_SPACER_CHAR, `[View Original Message](${link})`);
+					embed.setFooter(`Message sent at ${DateUtils.format(messageToPreview.createdAt)}`);
+					embed.setColor(messageToPreview.member?.displayColor || MEMBER_ROLE_COLOR);
 
 					callingMessage.channel.send(embed);
 				}
 			}
 		}
 	}
 
+	getChannelName(message: Message): String {
+		const textChannel = message.channel as TextChannel;
+
+		return textChannel.name;
+	}
+
+	escapeHyperlinks(content: String): String {
+		if (!content) {
+			return content;
```

Comment:
How can you return something that doesn't exist? A bit confused what's going on here.

Reply:
It's supposed to return `null`, `undefined` or `""` if content matches any of those, to prevent a `TypeError` when calling `content.replace()`. Maybe an elvis operator would be better? e.g `content?.replace()`